Dockerfiles/agent/entrypoint.d/otel-agent

```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+if [ -f /opt/datadog-agent/embedded/bin/otel-agent ]; then
```

Comment:
```suggestion
if [ -f /opt/datadog-agent/embedded/bin/otel-agent ] && [ -x /opt/datadog-agent/embedded/bin/otel-agent ]; then
```
I agree with the silent fail in that case as otherwise we would log everytime the container starts when we use an otel-less agent image.